plotlykt-core/src/jvmMain/kotlin/scientifik/plotly/FileExport.kt

```diff
@@ -1,25 +1,45 @@
 package scientifik.plotly
 
-import kotlinx.html.*
+import kotlinx.html.body
+import kotlinx.html.div
+import kotlinx.html.head
+import kotlinx.html.html
+import kotlinx.html.id
+import kotlinx.html.link
+import kotlinx.html.meta
+import kotlinx.html.script
 import kotlinx.html.stream.createHTML
+import kotlinx.html.title
+import kotlinx.html.unsafe
+import scientifik.plotly.assets.AssetsLocator
+import scientifik.plotly.assets.AssetsProvidingMode
+import scientifik.plotly.assets.of
 import java.awt.Desktop
 import java.io.File
 
 /**
  * Create a html string from plot
+ *
+ * @param provideAssets use this to control the way assets (js-files/images/etc)
+ *                      will be referenced. Check [specific entries][AssetsProvidingType]
+ *                      for details
  */
-fun Plot2D.makeHtml(): String {
+fun Plot2D.makeHtml(
+    provideAssets: AssetsProvidingMode = AssetsProvidingMode.Online
+): String {
     val tracesParsed = data.toJsonString()
     val layoutParsed = layout.toJsonString()
 
+    val a = AssetsLocator.of(provideAssets)
+
     return createHTML().html {
         head {
             meta {
                 charset = "utf-8"
-                script { src = "https://cdn.plot.ly/plotly-latest.min.js" }
+                script { src = a("https://cdn.plot.ly/plotly-latest.min.js") }
```

Comment:
I do not like the idea of inlining everything inside the plot because of its size. I suggest to create `.plotly` directory in the file location or in user root and place plotly files there. This way we can create a single asset directory for all plots. Also we can pre-bundle those files and avoid fetching at all, just check if asset directory exists. We can later add a small JS fragment which checks if asset exists and downloads them if needed.

Reply:
We can implement both options (e.g. `Offline` does exactly that).